whitesource/component.py

```diff
@@ -84,3 +89,136 @@ def download_component(
                 filtered_out_files += 1
 
     logger.info(f'{files_to_scan=}, {filtered_out_files=}')
+
+
+def apply_filters(
+    components: typing.Generator[typing.Union[tuple, typing.Any], typing.Any, typing.Any],
+    filters: typing.List[whitesource.model.WhiteSourceFilterCfg],
+) -> typing.Generator[typing.Union[tuple, typing.Any], typing.Any, typing.Any]:
+
+    component_filter = None
+    source_filter = None
+    resource_filter = None
+
+    for f in filters:
+        if f.type == 'component':
+            component_filter = f
+        elif f.type == 'source':
+            source_filter = f
+        elif f.type == 'resource':
+            resource_filter = f
+        else:
+            raise NotImplementedError
+
+    if component_filter:
+        components = _apply_component_filter(filter=component_filter, components=components)
+
+    if source_filter:
+        components = _apply_artifact_filter(filter=source_filter, components=components)
+
+    if resource_filter:
+        components = _apply_artifact_filter(filter=resource_filter, components=components)
```

Comment:
wouldn't later invocations override the results of previous ones, if more than one filter is defined?

Also, this function in general is doing this in quite a roundabout way. I'd suggest to call the appropriate function per filter type and aggregate all the results before moving on to the next filter type.